framework/src/vectorpostprocessors/SidesetInfoVectorPostprocessor.C

```diff
@@ -0,0 +1,192 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SidesetInfoVectorPostprocessor.h"
+#include <limits>
+
+#include "libmesh/quadrature.h"
+
+registerMooseObject("MooseApp", SidesetInfoVectorPostprocessor);
+
+template <>
+InputParameters
+validParams<SidesetInfoVectorPostprocessor>()
+{
+  InputParameters params = validParams<SideVectorPostprocessor>();
+  MultiMooseEnum meta_data_types("centroid=0 min=1 max=2 area=3");
+  params.addParam<MultiMooseEnum>(
+      "meta_data_types", meta_data_types, "Data types that are obtained and written to file.");
+  params.addClassDescription("This VectorPostprocessor collects meta data for provided sidesets.");
+  return params;
+}
+
+SidesetInfoVectorPostprocessor::SidesetInfoVectorPostprocessor(const InputParameters & parameters)
+  : SideVectorPostprocessor(parameters),
+    _meta_data_types(getParam<MultiMooseEnum>("meta_data_types")),
+    _sideset_ids(declareVector("Boundary IDs"))
+{
+  // this sets up the _vpp_entry_names vector
+  for (unsigned int j = 0; j < _meta_data_types.size(); ++j)
+  {
+    if (_meta_data_types[j] == "centroid" || _meta_data_types[j] == "min" ||
+        _meta_data_types[j] == "max")
+    {
+      for (unsigned int d = 0; d < _mesh.dimension(); ++d)
+      {
+        std::stringstream ss;
+        ss << _meta_data_types[j] << "_";
+        if (d == 0)
+          ss << "x";
+        else if (d == 1)
+          ss << "y";
+        else if (d == 2)
+          ss << "z";
+        _vpp_entry_names.push_back(ss.str());
+      }
+    }
+    else
+      _vpp_entry_names.push_back(_meta_data_types[j]);
+  }
+}
+
+void
+SidesetInfoVectorPostprocessor::initialize()
+{
+  for (auto & e : boundaryIDs())
+    _boundary_data[e] = BoundaryData();
+
+  // resize the sideset id vector
+  _sideset_ids.resize(numBoundaryIDs());
+
+  // now we can initialize the _meta_data vector
+  _meta_data.resize(_vpp_entry_names.size());
+  for (unsigned int j = 0; j < _vpp_entry_names.size(); ++j)
```

Comment:
Minor: Avoid using unsigned int for vector indices. We are starting to run into issues with some of these in the framework when we scale up to very large problems. I realize that that won't be an issue in either of the places you put them in this PR, but it's just something to watch for. You can use `std::size_t` or `MooseIndex(var)` to get the right types.